### PR TITLE
better collection ordering

### PR DESCRIPTION
**Pull request by jamesgorrie**
Obviously don't want to order by the path head, more the name of the folder.
Also learnt you can have multiple conditions with tuple instead of multiple `sortBy`s.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Tested, merging as breaks nothing.

File: collections/app/model/Node.scala [hunk 1]
```diff
@@ -22,8 +22,7 @@ object Node {
           // use the T at this level or an empty node to hold children
           Node(currentSlug, loop(children, thisFullPath), thisFullPath, thisLevel.headOption)
         }
-        .sortBy(_.fullPath.head)
-        .sortBy(_.children.isEmpty)
+        .sortBy(node => (node.children.isEmpty, node.basename))
```
Comment (theefer) [inline collections/app/model/Node.scala]: Not sure I understand this..?

Comment (jamesgorrie) [inline collections/app/model/Node.scala]: [I sort of kind of do](https://joelneely.wordpress.com/2008/03/29/sorting-by-schwartzian-transform-in-scala/).

In short, `sortBy` has a nice little trick built in when you sort by a tuple.
[Here's a simpler, practical example](http://stackoverflow.com/questions/10027543/how-to-sort-a-list-in-scala-by-two-fields).

If you're wandering about the actual reason for the parameters, it's sorted by if the collections has children, and then alphabetically, similar to file system UIs. So the collapsable ones are always at the top.

Slightly ironic screengrab:
nochildren

Comment (theefer) [inline collections/app/model/Node.scala]: Ah OK, wasn't sure why the No children thing. Is it sorting recursively down the tree too?

Comment (jamesgorrie) [inline collections/app/model/Node.scala]: It does as we're [sorting within the loop](https://github.com/guardian/grid/blob/e15305800f96db81f9295f55dc00b5e8811b8859/collections/app/model/Node.scala#L11-L26).

Comment (theefer) [inline collections/app/model/Node.scala]: Ah right cool!

**Pull request merged**